packages/jest-mock/src/index.ts

```diff
@@ -995,7 +995,7 @@ export class ModuleMocker {
   isMockFunction<T, Y extends Array<unknown> = Array<unknown>>(
     fn: (...args: Y) => T,
   ): fn is Mock<T, Y>;
-  isMockFunction(fn: unknown): boolean;
+  isMockFunction(fn: unknown): fn is Mock<unknown>;
```

Comment:
@SimenB Might be this is better idea. See tests. Thanks for good questions!

Reply:
perfect, thanks!